### DISCUSSION TITLE
brew stopped update/upgrade after upgrading to big sur

### DISCUSSION THREAD

**Discussion by eSKond** [Homebrew/discussions#190] [Everyday usage]
"brew upgrade" says:
Updating Homebrew...
Warning: You are using macOS 11.0.
We do not provide support for this pre-release version.
You will encounter build failures with some formulae.
Please create pull requests instead of asking for help on Homebrew's GitHub,
Discourse, Twitter or IRC. You are responsible for resolving any issues you
experience while you are running this pre-release version.

Why does it think it's pre-release?

"brew update" says:
fatal: unable to access 'https://github.com/Homebrew/brew/': Failed to connect to github.com port 443: Bad file descriptor
fatal: unable to access 'https://github.com/Homebrew/homebrew-cask/': Failed to connect to github.com port 443: Bad file descriptor
fatal: unable to access 'https://github.com/Homebrew/homebrew-core/': Failed to connect to github.com port 443: Bad file descriptor
fatal: unable to access 'https://github.com/Homebrew/homebrew-services/': Failed to connect to github.com port 443: Bad file descriptor
Error: Fetching /usr/local/Homebrew failed!
Fetching /usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask failed!
Fetching /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core failed!
Fetching /usr/local/Homebrew/Library/Taps/homebrew/homebrew-services failed!

What? Why?

Here is "brew config" output:
HOMEBREW_VERSION: 2.5.8-132-g29160b9
ORIGIN: https://github.com/Homebrew/brew
HEAD: 29160b9e7f07ab616aedcbdc95f2aeb2bc798a3b
Last commit: 3 weeks ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 3e5171c23807ebee3204c69a4f23e8be17548a24
Core tap last commit: 3 weeks ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.3a8sssmMFe/org.macosforge.xquartz:0
HOMEBREW_MAKE_JOBS: 4
Homebrew Ruby: 2.6.3 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: quad-core 64-bit haswell
Clang: 12.0 build 1200
Git: 2.29.2 => /usr/local/bin/git
Curl: 7.64.1 => /usr/bin/curl
Java: 1.8.271.09, 1.8.0_121, 1.7.0_51
macOS: 11.0.1-x86_64
CLT: 12.2.0.0.1.1604076827
Xcode: 12.2
XQuartz: 2.7.9 => /opt/X11

What to do? How to fix?

**Comment by SMillerDev** [comment]
> fatal: unable to access 'https://github.com/Homebrew/brew/': Failed to connect to github.com port 443: Bad file descriptor
fatal: unable to access 'https://github.com/Homebrew/homebrew-cask/': Failed to connect to github.com port 443: Bad file descriptor
fatal: unable to access 'https://github.com/Homebrew/homebrew-core/': Failed to connect to github.com port 443: Bad file descriptor
fatal: unable to access 'https://github.com/Homebrew/homebrew-services/': Failed to connect to github.com port 443: Bad file descriptor

This means there is an issue with your network connection to `github.com` this isn't likely related to Big Sur or homebrew.

**Reply by eSKond** [reply]
Of course i checked connection and no problem found with it:

> telnet github.com 443
Trying 140.82.121.3...
Connected to github.com.

**Reply by eSKond** [reply]
Just to be sure there is no any connection i run following command and it works:
> brew reinstall bundletool
Warning: You are using macOS 11.0.
We do not provide support for this pre-release version.
You will encounter build failures with some formulae.
Please create pull requests instead of asking for help on Homebrew's GitHub,
Discourse, Twitter or IRC. You are responsible for resolving any issues you
experience while you are running this pre-release version.

==> Downloading https://gist.githubusercontent.com/raw/ca85ede7ac072a44f48c658be55ff0d3/sample.aab
Already downloaded: /Users/sergey/Library/Caches/Homebrew/downloads/d453ac6ab61df1ca2f670e5d5d3e87f57a332dc2cdd618f74e6963310c12d3a6--sample.aab
==> Downloading https://github.com/google/bundletool/releases/download/1.3.0/bundletool-all-1.3.0.jar
Already downloaded: /Users/sergey/Library/Caches/Homebrew/downloads/659682cdd7e6d5f39a7a5c6e882c3ffb6adc5b18d4b40a7d95f42f191699a787--bundletool-all-1.3.0.jar
==> Reinstalling bundletool 
🍺  /usr/local/Cellar/bundletool/1.3.0: 4 files, 23.6MB, built in 6 seconds

**Comment by thepaul** [comment]
I had the same problem today on Big Sur, and the problem turned out to be Little Snitch. Somehow it had decided `brew.sh` should no longer be allowed to make connections to github.com. It's possible this was due to me mis-clicking something in a Little Snitch dialog popup. OP might want to check that.

**Reply by BahodurSaidov** [reply]
Similarly, I had to disable LuLu .
